### ISSUE TITLE
Visualization - default graph shows irrelevant x axes for some projects

### ISSUE THREAD

**Issue by omri374** [comet-ml/issue-tracking#99]
Hi,
First, thanks for all your hard work! Comet.ml is an awesome tool!

I have a project with many experiments, all have multiple parameters and multiple metrics. The default visualization lets me create a line chart, but with predefined X axis values (duration, wall, step), which I'm not sure what they mean in my setting. I get some hex numbers which I'm not sure what they mean. (Experiment key?). See attached screenshot.

This is what I currently get:
![image](https://user-images.githubusercontent.com/3776619/44681133-698cc300-aa48-11e8-9bed-154ef40d2e28.png)

Maybe it's just my misunderstanding. I'm not sure what would be a better representation in this case, but I found this option to be irrelevant for my scenario.

**Comment by neighthan** [comment]
The labels in the legend are indeed experiment keys. I have an issue open (#97) to make it easier to replace these with model names instead.

As for the x-axis, what would you want that to be? wall means the wall time when the metric was logged (e.g. if you started a model at 10 am, you might have a curve with your logged metrics from then until 11 am, if your model trains for an hour).

duration means how far into the model training you are. Using the same example, your curve would now be from 0 hours to 1 hour for the above model. If you want to see which model reaches best performance the fastest, using duration would be easier than wall time (so that all of the models start at the same time, duration of 0, instead of at whatever time you actually ran them).

When you log a metric, you can also specify a step for it (e.g. `experiment.log_metric('acc', 0.8, step=100)`). Step can be good for an x axis when you want to compare performance of models without penalizing slower ones like duration will do.

step doesn't always apply (because you have to have logged it with your metric to be able to use it), but you'll always have a wall time and a duration when you log any metric, so those apply to any project.

(note that the line chart only makes sense for metrics that you log multiple times during an experiment; if this is a value you just compute once at the end, you might want to look at the bar chart instead)

**Comment by ceceshao1** [comment]
Believe this is related to #102 @omri374? 

The experiment key, which as @neighthan pointed out, are the hex numbers.

Thanks for an excellent response @neighthan! Said a lot of things I was going to cover and provided a great explanation around step, duration, and wall. @omri374 - if you have any further questions, please don't hesitate to ask!

**Comment by omri374** [comment]
Thanks for the great response @neighthan! I now better understand what this chart is about.

Regarding:

> (note that the line chart only makes sense for metrics that you log multiple times during an experiment; if this is a value you just compute once at the end, you might want to look at the bar chart instead)
I see what you're saying, this really isn't the scenario I'm working on.

As a feature request, as someone who isn't running deep learning (or long running) models, I think that a great feature would be a line chart or scatter plot that could give this functionality: plot the values of one parameter (x axis) vs. values of one metric (y axis). It doesn't always make sense as there are many other parameters that might affect the metric value, but it does provide some intuition on whether a certain parameter affects a metric or not.
For example: plot learning rate values vs. MSE. If learning rate values are uncorrelated with MSE (as seen on such graph), I should probably invest less on this hyperparameter and focus on others.
Just a thought. keep up the good work!

**Comment by dsblank** [comment]
There is now a scatter plot available in the Project level view. Further enhancements are in development.

**Comment by dsblank** [comment]
Now, you can make your own Plots, using whatever you wish for x, y, z axis's, and create your own legends. Please see: https://www.comet.ml/docs/javascript-sdk/getting-started/
